helix-term/src/commands/lsp.rs

```diff
@@ -290,8 +296,28 @@ fn diag_picker(
         warning: cx.editor.theme.get("warning"),
         error: cx.editor.theme.get("error"),
     };
+    let mut informations = 0;
+    let mut warnings = 0;
+    let mut errors = 0;
+    let mut hints = 0;
+
+    flat_diag
+        .iter()
+        .filter_map(|i| i.diag.severity)
+        .for_each(|severity| match severity {
+            DiagnosticSeverity::INFORMATION => informations += 1,
+            DiagnosticSeverity::WARNING => warnings += 1,
+            DiagnosticSeverity::ERROR => errors += 1,
+            DiagnosticSeverity::HINT => hints += 1,
+            _ => (),
+        });
 
+    let picker_title = format!(
+        "Diagnostics: [{} E] [{} W] [{} I] [{} H]",
```

Comment:
I think just: `{} E, {} W, {} I, {} H` will be enough and have less sigils to read. Commas are easier to skip than `[]` too